Assets/MixedRealityToolkit.SDK/Inspectors/UX/Interactable/PressableButtonInspector.cs

```diff
@@ -253,8 +259,60 @@ public override void OnInspectorGUI()
         {
             serializedObject.Update();
 
+            // Ensure there is a touchable.
+            if (touchable == null)
+            {
+                EditorGUILayout.HelpBox($"{target.GetType().Name} requires a {nameof(NearInteractionTouchableSurface)}-derived component on this game object to function.", MessageType.Warning);
+
+                bool isUnityUI = (button.GetComponent<RectTransform>() != null);
+                var typeToAdd = isUnityUI ? typeof(NearInteractionTouchableUnityUI) : typeof(NearInteractionTouchable);
+
+                if (GUILayout.Button($"Add {typeToAdd.Name} component"))
+                {
+                    Undo.RecordObject(target, string.Concat($"Add {typeToAdd.Name}"));
+                    var addedComponent = button.gameObject.AddComponent(typeToAdd);
+                    touchable = (NearInteractionTouchableSurface)addedComponent;
+                }
+                else
+                {
+                    // It won't work without it, return to avoid nullrefs.
+                    return;
+                }
+            }
+
+            // Ensure that the touchable has EventsToReceive set to Touch
```

Comment:
Do you also need to check if there is a pressablebuttoneventrouter? Most pressable buttons use Interactable, which need a pressable button event router. Maybe check that case as well?

Reply:
I haven't attempted to add one in code.  I don't think it would be very straightforward, but I think it would be possible.

Regarding the EventRouter (PhysicalPressEventRouter or SubmitEventRouter), I don't think we should warn for it.  It's not strictly required and users of custom button classes won't want to use one.